pandas/core/internals/array_manager.py

```diff
@@ -526,9 +557,17 @@ def copy_func(ax):
 
         if deep:
             new_arrays = [arr.copy() for arr in self.arrays]
+            refs = None
         else:
-            new_arrays = self.arrays
-        return type(self)(new_arrays, new_axes)
+            new_arrays = list(self.arrays)
+            refs = [weakref.ref(arr) for arr in self.arrays]
+
+        # error: Argument 3 to "BaseArrayManager" has incompatible type
+        # "Optional[List[ReferenceType[Union[ndarray[Any, Any], ExtensionArray]]]]";
+        # expected "Optional[List[Optional[ReferenceType[Any]]]]" [arg-type]
+        return type(self)(
+            new_arrays, new_axes, refs, verify_integrity=False  # type: ignore[arg-type]
```

Comment:
Anybody an idea why mypy finds those incompatible? To me it seems the expected type is just more broad, and the argument is a strict subset of it, which I thought would be OK (cc @simonjayhawkins)

Reply:
just guessing: L570 try `refs: list[weakref.ref | None] = [...]`?

Reply:
Thanks, will try (to be honest, I find this a bug in mypy (or very annoying feature) that it cannot see that `list[type]` is strictly a subset of `list[type | None]`

Reply:
Using something like Sequence is the only way around this unfortunately